dotnet/src/SemanticKernel.Core/Functions/KernelPlugin.cs

```diff
@@ -80,6 +79,7 @@ public KernelPlugin(string name, string? description, IEnumerable<KernelFunction
     public void AddFunction(KernelFunction function)
     {
         Verify.NotNull(function);
+        function.Metadata.PluginName = this.Name;
```

Comment:
This seems problematic. What happens if the same KernelFunction is added to two different plugins?

Reply:
@matthewbolanos The issue #3855 here is that when we add functions to a plugin the function meteadata doesn't get assigned the plugin name i.e. it only happens if you call `GetFunctionsMetadata` and this is an extension method.

The proposed fix would be to set the plugin name when the function is added to the plugin but this causes the issue Stephen identified. So we can

1. Leave things as they are but then the function metadata is inconsistent
2. Prevent the same function being added to two plugins i.e. throw an exception if someone attempts this

Which behaviour should we aim for?

Reply:
Let’s stick with option 1. I can then call it out in docs as a limitation. The preferred route for calling functions should probably be through the `string pluginName, string functionName` `InvokeAsync` method since we get the plugin name at that point.

can we confirm the metadata is populated for that code path?